### PR TITLE
suggestions for PR  bbc#144

### PR DESCRIPTION
**Pull request by pietrop**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

[https://github.com/bbc/react-transcript-editor/pull/144](https://github.com/bbc/react-transcript-editor/pull/144)

**Describe what the PR does**    

some changes to show suggestions for PR https://github.com/bbc/react-transcript-editor/pull/144 around edge case of trying to preserve block level time-codes and speaker informations when  user paste accurate text into the editor.

**State whether the PR is ready for review or whether it needs extra work**    

Ready for review

**Additional context**    

Just to demonstrate, show, and discuss ideas.

### CODE REVIEW

**Comment by pietrop** [comment]
## Handling edge case: pasting accurate text in editor
This PR shows a possible way for handling edge case when pasting text for retaining paragraph timecode and speaker name
- [x] handling timecode at paragraph level when corrected text is pasted into the editor
- [x] handling speaker names at paragraph level by using default `UKN` speaker label, which is good if replacing everything, not so good if replacing a line inside a paragraph by  pasting in the correct text (as we might want to keep the speaker name in that case).

## speaker name 
 If you export draftJs json from the demo app, you can see that in the paragraph block there is a `data.speaker` attribute.  

```js
    {
      "key": "129ia",
      "text": "he'd ordered and I was really excited about it because I've always loved about this one has really caught technical features. It had more orders and touch sensors. It had an infra red camera and one of the things that had was a tilt sensor so it. Knew what direction. It was facing. If and when you held it upside down.",
      "type": "paragraph",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [
        {
          "offset": 0,
          "length": 4,
          "key": 29
        },
…..
      ],
      "data": {
        "speaker": "F_S12",
        "words": [
          {
            "start": 24.65,
            "confidence": 0.59,
            "end": 24.84,
            "word": "he'd",
            "punct": "he'd",
            "index": 29,
            "speaker": "F_S12"
          },
```


in [`createEntity`](https://github.com/bbc/react-transcript-editor/pull/144/files#diff-9e2a4053c520c0646c7f5bd7a6167d96R17) (or a bit cleaner, in another function as a second pass) you might need to add speaker names so that you can access it in [`createContentFromEntityList`](https://github.com/bbc/react-transcript-editor/pull/144/files#diff-9e2a4053c520c0646c7f5bd7a6167d96R28)

Not sure if start is under [`block.data.start`](https://github.com/bbc/react-transcript-editor/pull/144/files#diff-9e2a4053c520c0646c7f5bd7a6167d96R44) in [`createContentFromEntityList`](https://github.com/bbc/react-transcript-editor/pull/144/files#diff-9e2a4053c520c0646c7f5bd7a6167d96R28)

**TL;DR**: See [the diffs in this PR](https://github.com/murezzda/react-transcript-editor/pull/1/files) for some suggestions around possible ways to handle edge case when user pastes text for retaining paragraph timecode and speaker name.

**Comment by pietrop** [comment]
Another possible solution is to catch the pasted text further upstream and to use [`handlePastedText`](https://draftjs.org/docs/api-reference-editor.html#handlepastedtext) also mentioned in [this article about race condition](https://draftjs.org/docs/advanced-topics-editorstate-race-conditions) as well as [this comment](https://github.com/facebook/draft-js/issues/416#issuecomment-221639163).

The hunch being that when you paste some text into the editor, it is missing the `entities` so draftjs does not identify it as words, and therefore speaker and time info are missing.
